scipy/optimize/_linprog.py

```diff
@@ -554,6 +554,7 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
 
     lp = _LPProblem(c, A_ub, b_ub, A_eq, b_eq, bounds, x0)
     lp, solver_options = _parse_linprog(lp, options)
+    tol = solver_options.get('tol', 1e-9)
```

Comment:
Moved this up from below. For HiGHS, `tol` is not an option, so users will not really be able to adjust this. I don't know whether it's worth exposing `tol` just for the sanity check, and I'm not sure whether we should use one of the other HiGHS-specific tolerances.